src/index.js

```diff
@@ -44,6 +46,22 @@ import { Environment } from './environment';
     config: config.helpGuide,
   });
 
+  // hide content if a browser is not supported
+  const browser = browserDetails();
+  const appendMessage = () => {
+    document.getElementById('qg-content').innerHTML = errorMessages.browserNotSupported();
+  }
+
+  if((browser.name === "Firefox" && browser.version <= 80) || (browser.name === "IE" && browser.version <= 11) || (browser.name === "Edge" && browser.version < 17)  ||  (browser.name === "Safari" && browser.version < 11)){
+    appendMessage();
+  }
+  if(browser.name === "Chrome" && browser.version <= 84 && !navigator.userAgent.match(/SAMSUNG|SGH-[I|N|T]|GT-[I|P|N]|SM-[N|P|T|Z|G]|SHV-E|SCH-[I|J|R|S]|SPH-L/i)){
```

Comment:
Can you explain to  me what this does please I am struggling to follow it.

Reply:
Samsung browser uses Chrome behind the scenes but they have modified it to an extent that they are not following Chrome versioning (they do not have 84 etc, they have a separate versioning system) so in the above line of code I am ignoring Chrome running on the Samsung browser.

Reply:
Normally I suggest that to ensure that this code is maintainable that it be broken into functions. More than two if statements is where I will always suggest to have it broken into functions. As readable code is maintainable code. However, if you could at least comment on the code it will help in the future.